pkg/kubelet/rkt/rkt.go

```diff
@@ -142,3 +149,253 @@ func (r *Runtime) runCommand(args ...string) ([]string, error) {
+		addCaps = getAllCapabilities()
+	} else {
+		addCaps = getCapabilities(c.Capabilities.Add)
+	}
+	if len(addCaps) > 0 {
+		// TODO(yifan): Replace with constructor, see:
+		// https://github.com/appc/spec/issues/268
+		isolator := appctypes.Isolator{
+			Name:     "os/linux/capabilities-retain-set",
+			ValueRaw: rawValue(fmt.Sprintf(`{"set":[%s]}`, addCaps)),
+		}
+		app.Isolators = append(app.Isolators, isolator)
+	}
+
+	// Removed capabilities.
+	dropCaps := getCapabilities(c.Capabilities.Drop)
+	if len(dropCaps) > 0 {
+		// TODO(yifan): Replace with constructor, see:
+		// https://github.com/appc/spec/issues/268
+		isolator := appctypes.Isolator{
+			Name:     "os/linux/capabilities-remove-set",
+			ValueRaw: rawValue(fmt.Sprintf(`{"set":[%s]}`, dropCaps)),
+		}
+		app.Isolators = append(app.Isolators, isolator)
+	}
+
+	// Resources.
+	resources := make(map[api.ResourceName]resource)
+	for name, quantity := range c.Resources.Limits {
+		resources[name] = resource{limit: quantity.String()}
+	}
+	for name, quantity := range c.Resources.Requests {
+		r, ok := resources[name]
+		if !ok {
+			r = resource{}
+		}
+		r.request = quantity.String()
+		resources[name] = r
+	}
+	var acName appctypes.ACName
+	for name, res := range resources {
+		switch name {
+		case api.ResourceCPU:
+			acName = "resource/cpu"
+		case api.ResourceMemory:
+			acName = "resource/memory"
+		default:
+			return fmt.Errorf("resource type not supported: %v", name)
+		}
+		// TODO(yifan): Replace with constructor, see:
+		// https://github.com/appc/spec/issues/268
+		isolator := appctypes.Isolator{
+			Name:     acName,
+			ValueRaw: rawValue(fmt.Sprintf(`{"request":%q,"limit":%q}`, res.request, res.limit)),
+		}
+		app.Isolators = append(app.Isolators, isolator)
+	}
+	return nil
+}
+
+// setApp overrides the app's fields if any of them are specified in the
+// container's spec.
+func setApp(app *appctypes.App, c *api.Container) error {
+	// Override the exec.
+	// TOOD(yifan): Revisit this for the overriding rule.
+	if len(c.Command) > 0 || len(c.Args) > 0 {
+		app.Exec = append(c.Command, c.Args...)
+	}
+
+	// TODO(yifan): Use non-root user in the future, see:
+	// https://github.com/coreos/rkt/issues/820
+	app.User, app.Group = "0", "0"
+
+	// Override the working directory.
+	if len(c.WorkingDir) > 0 {
+		app.WorkingDirectory = c.WorkingDir
+	}
+
+	// Override the environment.
+	// TODO(yifan): Use RunContainerOptions.
+	if len(c.Env) > 0 {
+		app.Environment = []appctypes.EnvironmentVariable{}
+	}
+	for _, env := range c.Env {
+		app.Environment = append(app.Environment, appctypes.EnvironmentVariable{
+			Name:  env.Name,
+			Value: env.Value,
+		})
+	}
+
+	// Override the mount points.
+	if len(c.VolumeMounts) > 0 {
+		app.MountPoints = []appctypes.MountPoint{}
+	}
+	for _, m := range c.VolumeMounts {
+		mountPointName, err := appctypes.NewACName(m.Name)
+		if err != nil {
+			glog.Errorf("rkt: Cannot use the volume mount's name %q as ACName: %v", m.Name, err)
```

Comment:
I'd argue we shouldn't log here (and below) since we are returning the error. We can make a new error with more data and let higher layers log if they'd like